bridge-proxy/src/bridge-proxy.rs

```diff
@@ -75,36 +67,37 @@ pub trait BridgeProxyContract:
 
     fn refund_transaction(&self, tx_id: usize) {
         let tx = self.get_pending_transaction_by_id(tx_id);
+        let esdt_safe_addr = self.esdt_safe_address().get();
 
-        let _: IgnoreValue = self
-            .get_esdt_safe_proxy_instance()
+        self.tx()
+            .to(esdt_safe_addr)
+            .typed(esdt_safe_proxy::EsdtSafeProxy)
             .create_transaction(tx.from)
-            .with_esdt_transfer((tx.token_id.clone(), 0, tx.amount.clone()))
-            .execute_on_dest_context();
+            .egld_or_single_esdt(&EgldOrEsdtTokenIdentifier::esdt(tx.token_id), 0, &tx.amount)
```

Comment:
this could only be .single_esdt(..) to avoid creating a EgldOrEsdtTokenIdentifier